src/coreclr/src/tools/r2rtest/Commands/CompileSerpCommand.cs

```diff
@@ -0,0 +1,188 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Globalization;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace R2RTest
+{
+    class CompileSerpCommand
+    {
+        public static int CompileSerpAssemblies(BuildOptions options)
+        {
+            if (options.InputDirectory == null)
+            {
+                Console.Error.WriteLine("Specify --response-file or --input-directory containing multiple response files.");
+                return 1;
+            }
+
+            if (options.CoreRootDirectory == null)
+            {
+                Console.Error.WriteLine("--core-root-directory (--cr) is a required argument.");
+                return 1;
+            }
+
+
+            // This command does not work in the context of an app, just a loose set of rsp files so don't execute anything we compile
+            options.NoJit = true;
+            options.NoEtw = true;
+
+            string serpDir = options.InputDirectory.FullName;
+            if (!File.Exists(Path.Combine(serpDir, "runserp.cmd")))
+            {
+                Console.Error.WriteLine($"Error: InputDirectory must point at a SERP build. Could not find {Path.Combine(serpDir, "runserp.cmd")}");
+                return 1;
+            }
+
+            string whiteListFilePath = Path.Combine(serpDir, "WhitelistDlls.txt");
+            if (!File.Exists(whiteListFilePath))
+            {
+                Console.Error.WriteLine($"File {whiteListFilePath} was not found");
+                return 1;
+            }
+
+            if (!File.Exists(Path.Combine(options.AspNetPath.FullName, "Microsoft.AspNetCore.dll")))
+            {
+                Console.Error.WriteLine($"Error: Asp.NET Core path must contain Microsoft.AspNetCore.dll");
+                return 1;
+            }
+
+            // Add all assemblies from the various SERP packages (filtered by ShouldInclude)
+            List<string> binFiles = Directory.GetFiles(Path.Combine(serpDir, "App_Data\\Answers\\Services\\Packages"), "*.dll", SearchOption.AllDirectories).ToList();
+
+            binFiles = binFiles.Where((string x) => ShouldInclude(x)).ToList();
+
+            // Add a whitelist of assemblies from bin
+            string binDir = Path.Combine(serpDir, "bin");
+            foreach (string item in new HashSet<string>(File.ReadAllLines(whiteListFilePath)))
```

Comment:
Is the whitelist always expected to exist? If not, I guess you're going to get a <code>FileNotFoundException</code> here we should probably tolerate.

Reply:
Yes, this is part of the drop and is always there.